### PR TITLE
Clean up ui-components styles

### LINKED ISSUE #6043: Unset body color
I think we shouldn't explicitly set the body color and theme:

https://github.com/jupyterlab/jupyterlab/blob/b212097539d3725bb46d1f7b92bb4c93db9827a3/packages/ui-components/style/index.css#L18-L19

This can mess up packages that end up inheriting from a body font color that is black, especially in a dark theme. Why don't we unset these like we do with other things in this override file?

CC @gnestor

### PR DESCRIPTION
**Pull request by gnestor**
Fixes #6043

This removes some unnecessary overrides and documents what overrides what. This doesn't affect the style of the app at all.

### CODE REVIEW

**Comment by jasongrout** [comment]
Thanks! I think we *do* need to override those global body styles that blueprint sets. I was thinking we should do something like:

```
body {
  color: unset;
  font-family: var(--jp-ui-font-family);
}
```

**Comment by gnestor** [comment]
Oh, gotcha. I just updated and tested and it looks good 👍

File: packages/ui-components/style/index.css [hunk 1]
```diff
@@ -76,8 +75,6 @@ a:hover {
 }
 
 .jp-InputGroup input:focus {
-  /* box-shadow: inset 0 0 0 var(--jp-border-width) var(--jp-input-active-border-color),
```
Comment (jasongrout) [inline packages/ui-components/style/index.css]: One more question: is there a reason we aren't using theme colors for this focus highlight, like in the commented code? Any time I see hardcoded colors like this, I wonder how it behaves in different themes.

Comment (gnestor) [reply packages/ui-components/style/index.css]: Yes yes, good catch!

**Comment by ellisonbg** [comment]
We shouldn't rely on inherited styles (such as setting the font face on the body) for these UI components, otherwise, they won't work in other contexts where the inherited styles are different.

File: packages/ui-components/style/index.css [hunk 2]
```diff
@@ -15,7 +15,7 @@ html {
 }
 
 body {
-  color: #182026;
+  color: unset;
   font-family: var(--jp-ui-font-family);
```
Comment (ellisonbg) [inline packages/ui-components/style/index.css]: We should remove this line - see comment about inherited styles in main comment thread.

Comment (ellisonbg) [reply packages/ui-components/style/index.css]: This change will likely require us to hunt through the UI and find all the case where the font family is not set properly though.

Comment (jasongrout) [reply packages/ui-components/style/index.css]: Note that jlab already sets that body style: https://github.com/jupyterlab/jupyterlab/blob/86ac08051d573af947c4a51554a85fd5abf96246/packages/application/style/index.css#L13-L19
I think you're implying that that css should not be in jlab either.

Comment (ellisonbg) [reply packages/ui-components/style/index.css]: Yes ;-)

Comment (jasongrout) [reply packages/ui-components/style/index.css]: If you consider this a 1.0 issue, would you mind opening a new issue for this, and tagging it UX and 1.0?

**Comment by jasongrout** [comment]
Thanks!

**Pull request merged**